### PR TITLE
Divyanshu | Test | Implementation of Kenya VAT report

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cv1ctb7

### CODE REVIEW

File: apps/web-giddh/src/app/services/vat.service.ts [hunk 1]
```diff
@@ -22,10 +22,11 @@ export class VatService {
      * @return {*}  {Observable<BaseResponse<any, any>>}
      * @memberof VatService
      */
-    public getCountryWiseVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW' = 'UK'): Observable<BaseResponse<any, any>> {
+    public getCountryWiseVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW' | 'KE' = 'UK'): Observable<BaseResponse<any, any>> {
         this.companyUniqueName = this.generalService.companyUniqueName;
-        const apiEndPoint = countryCode === 'UK' ? VAT_API.VIEW_REPORT : VAT_API.VIEW_ZW_REPORT;
-        let url = this.config.apiUrl + apiEndPoint;
+        const apiEndPoint = countryCode === 'ZW' ? VAT_API.VIEW_ZW_REPORT : (countryCode === 'KE' ? VAT_API.VIEW_KE_REPORT : VAT_API.VIEW_REPORT);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/vat.service.ts]:
```suggestion
        const apiEndPoint = countryCode === 'ZW' ? VAT_API.VIEW_ZW_REPORT : (countryCode === 'KE' ? VAT_API.VIEW_KENYA_REPORT : VAT_API.VIEW_REPORT);
```

File: apps/web-giddh/src/app/services/vat.service.ts [hunk 2]
```diff
@@ -70,9 +71,10 @@ export class VatService {
             }), catchError((e) => this.errorHandler.HandleCatch<any, any>(e, request)));
     }
 
-    public downloadVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW'): Observable<BaseResponse<any, any>> {
+    public downloadVatReport(request: VatReportRequest, countryCode: 'UK' | 'ZW' | 'KE'): Observable<BaseResponse<any, any>> {
         this.companyUniqueName = this.generalService.companyUniqueName;
-        const apiEndPoint = countryCode === 'UK' ? VAT_API.DOWNLOAD_REPORT : VAT_API.DOWNLOAD_ZW_REPORT;
+        const apiEndPoint = countryCode === 'ZW' ? VAT_API.DOWNLOAD_ZW_REPORT : (countryCode === 'KE' ? VAT_API.DOWNLOAD_KE_REPORT : VAT_API.DOWNLOAD_REPORT);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/vat.service.ts]:
```suggestion
        const apiEndPoint = countryCode === 'ZW' ? VAT_API.DOWNLOAD_ZW_REPORT : (countryCode === 'KE' ? VAT_API.DOWNLOAD_KENYA_REPORT : VAT_API.DOWNLOAD_REPORT);
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 3]
```diff
@@ -79,91 +79,142 @@
...
-                                        <table mat-table [dataSource]="report.sections"
+                                        <table mat-table [dataSource]="report.sections" [ngClass]="{'hide-last-row': isKenyaCompany}"
                                             class="table basic mr-t2 on-mobile-view vat-report-table">
                                             <ng-container matColumnDef="number">
-                                                <th mat-header-cell *matHeaderCellDef></th>
+                                                <th mat-header-cell *matHeaderCellDef>
+                                                    <ng-container *ngIf="isKenyaCompany"> {{ localeData?.box }}</ng-container>
+                                                </th>
                                                 <td mat-cell *matCellDef="let element" width="5%">
                                                     {{ element.section }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]: element?.section

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 4]
```diff
@@ -79,91 +79,142 @@
...
+                                                            (click)="viewVatReportTransactions(element.section)">{{
+                                                            element.description }}
+                                                        </a>
+                                                        <a href="javascript:void(0)" class="text-light invoice-preview ml-1"
+                                                            [matTooltip]="element.toolTip" [matTooltipPosition]="'above'">
+                                                            <span class="icon-info text-light"></span>
+                                                        </a>
+                                                    </ng-container>
+                                                    <ng-container *ngIf="isKenyaCompany">
+                                                        {{ element.description }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                        {{ element?.description }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 5]
```diff
@@ -79,91 +79,142 @@
...
+                                                    <ng-container *ngIf="!isKenyaCompany">
+                                                        {{ commonLocaleData?.app_amount }}
+                                                        ({{ activeCompany?.baseCurrency }})
+                                                    </ng-container>
                                                 </th>
                                                 <td mat-cell *matCellDef="let element" class="text-right" width="15%">
                                                     {{ activeCompany.baseCurrencySymbol }}
-                                                    {{ element.amount | giddhCurrency }}
+                                                    <ng-container *ngIf="!isKenyaCompany">
+                                                        {{ element.amount | giddhCurrency }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                        {{ element?.amount | giddhCurrency }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 6]
```diff
@@ -79,91 +79,142 @@
...
+                                                    </ng-container>
                                                 </th>
                                                 <td mat-cell *matCellDef="let element" class="text-right" width="15%">
                                                     {{ activeCompany.baseCurrencySymbol }}
-                                                    {{ element.vatAmount | giddhCurrency }}
+                                                    <ng-conatiner *ngIf="!isKenyaCompany">
+                                                        {{ element.vatAmount | giddhCurrency }}
+                                                    </ng-conatiner>
+                                                    <ng-conatiner *ngIf="isKenyaCompany">
+                                                        {{ (element.rate ?? 0) | giddhCurrency }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                        {{ (element?.rate ?? 0) | giddhCurrency }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 7]
```diff
@@ -79,91 +79,142 @@
...
+                                                    </ng-container>
+                                                    <ng-container *ngIf="isKenyaCompany">
+                                                        {{ localeData?.amountOfOutputTax }}
+                                                    </ng-container>
                                                 </th>
                                                 <td mat-cell *matCellDef="let element" class="text-right" width="20%">
                                                     {{ activeCompany.baseCurrencySymbol }}
-                                                    {{ element.adjustmentAmount | giddhCurrency }}
+                                                    <ng-container *ngIf="!isKenyaCompany">
+                                                        {{ element.adjustmentAmount | giddhCurrency }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                        {{ element?.adjustmentAmount | giddhCurrency }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 8]
```diff
@@ -79,91 +79,142 @@
...
+                                                    <ng-container *ngIf="isKenyaCompany">
+                                                        {{ activeCompany.baseCurrencySymbol }}
+                                                        {{ (element?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
+                                                    </ng-container>
                                                 </td>
                                             </ng-container>
-                                            <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
-                                            <tr mat-row *matRowDef="let row; columns: displayedColumns"></tr>
-                                            <tr mat-footer-row *matFooterRowDef="displayedColumns"></tr>
+                                            <tr mat-header-row *matHeaderRowDef="isKenyaCompany? keDisplayedColumns :  displayedColumns"></tr>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                            <tr mat-header-row *matHeaderRowDef="isKenyaCompany? kenyaDisplayedColumns :  displayedColumns"></tr>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 9]
```diff
@@ -79,91 +79,142 @@
...
+                                            <tr mat-row *matRowDef="let row; columns: isKenyaCompany? keDisplayedColumns :  displayedColumns"></tr>
+                                            <tr mat-footer-row *matFooterRowDef="isKenyaCompany? keDisplayedColumns :  displayedColumns"></tr>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                            <tr mat-footer-row *matFooterRowDef="isKenyaCompany? kenyaDisplayedColumns :  displayedColumns"></tr>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 10]
```diff
@@ -172,31 +223,61 @@
...
-                                                        {{ element.amount | giddhCurrency }}</strong>
+                                            <ng-container [matColumnDef]="!isKenyaCompany ? 'tooltip' : 'vat-amount'">
+                                                <th mat-header-cell *matHeaderCellDef [ngClass]="{'text-right': isKenyaCompany}">
+                                                    <ng-container *ngIf="isKenyaCompany">{{ localeData?.vat_amount }}</ng-container>
+                                                </th>
+                                                <td mat-cell *matCellDef="let element" [ngClass]="{'text-right': isKenyaCompany}">
+                                                    <ng-container *ngIf="!isKenyaCompany">
+                                                        <span class="pr-1">{{ element.toolTip }}</span>
+                                                        <strong>{{ activeCompany.baseCurrencySymbol }}
+                                                            {{ element.amount | giddhCurrency }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                                            {{ element?.amount | giddhCurrency }}
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 11]
```diff
@@ -172,31 +223,61 @@
...
+                                                </td>
+                                                <td mat-footer-cell *matFooterCellDef class="text-right font-weight-bold">
+                                                    {{ activeCompany.baseCurrencySymbol }}
+                                                    {{ (vatReport[i]?.sections[vatReport[i]?.sections?.length -
+                                                        1]?.totalAmountOfOutputTax ?? 0) | giddhCurrency }}
                                                 </td>
                                             </ng-container>
-                                            <tr mat-header-row *matHeaderRowDef="uaeDisplayedColumns"></tr>
-                                            <tr mat-row *matRowDef="let row; columns: uaeDisplayedColumns"></tr>
+                                            <tr mat-header-row *matHeaderRowDef="isKenyaCompany ? keDisplayedColumnsForLastSection : uaeDisplayedColumns"></tr>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                            <tr mat-header-row *matHeaderRowDef="isKenyaCompany ? kenyaDisplayedColumnsForLastSection : uaeDisplayedColumns"></tr>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.html [hunk 12]
```diff
@@ -172,31 +223,61 @@
...
+                                            <tr mat-row *matRowDef="let row; columns: isKenyaCompany?  keDisplayedColumnsForLastSection : uaeDisplayedColumns"></tr>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.html]:
```suggestion
                                            <tr mat-row *matRowDef="let row; columns: isKenyaCompany?  kenyaDisplayedColumnsForLastSection : uaeDisplayedColumns"></tr>
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 13]
```diff
@@ -86,11 +86,15 @@ export class VatReportComponent implements OnInit, OnDestroy {
     /** Hold Zimbabwe main table displayed columns */
     public zwDisplayedColumns: string[] = ['name', 'mat-code', 'vos-amount', 'vos-decimal', 'ot-amount', 'ot-decimal'];
     /** Hold Zimbabwe table header row displayed columns */
-    public zwDisplayedHeaderColumns = ['section','office-use', 'value-of-supply', 'output-tax'];
+    public zwDisplayedHeaderColumns = ['section', 'office-use', 'value-of-supply', 'output-tax'];
     /** Hold Zimbabwe table displayed columns for last section */
-    public zwDisplayedColumnsForLastSection: string[] =  ['name', 'amount', 'decimal'];
+    public zwDisplayedColumnsForLastSection: string[] = ['name', 'amount', 'decimal'];
+    /** Hold Kenya table displayed columns */
+    public keDisplayedColumns: string[] = ['number', 'description', 'amount', 'rate', 'ot-amount'];
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]:
```suggestion
    public kenyaDisplayedColumns: string[] = ['number', 'description', 'amount', 'rate', 'ot-amount'];
```

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 14]
```diff
@@ -86,11 +86,15 @@ export class VatReportComponent implements OnInit, OnDestroy {
...
+    /** Hold Kenya table displayed columns */
+    public keDisplayedColumnsForLastSection: string[] = ['number', 'description', 'vat-amount'];
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]:
```suggestion
    public kenyaDisplayedColumnsForLastSection: string[] = ['number', 'description', 'vat-amount'];
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/vat.service.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/vat.service.ts]: Done

File: apps/web-giddh/src/app/services/vat.service.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/vat.service.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 9 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 10 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 11 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.html (hunk 12 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.html]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 13 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 14 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**